packages/pages-shared/src/asset-server/rulesEngine.ts

```diff
@@ -67,11 +83,15 @@ export const generateRulesMatcher = <T>(
 	][];
 
 	return ({ request }: { request: Request }) => {
-		const { pathname, host } = new URL(request.url);
+		const { pathname, host, search, searchParams } = new URL(request.url);
+		searchParams.sort()
+		const sortedSearch = searchParams.toString()
 
 		return compiledRules
 			.map(([{ crossHost, regExp }, match]) => {
-				const test = crossHost ? `https://${host}${pathname}` : pathname;
+				const test = crossHost
+					? `https://${host}${pathname}${sortedSearch}`
+					: `${pathname}${sortedSearch.length > 0 ? sortedSearch : ""}`;
```

Comment:
This line that was added is unnecessary, `sortedSearch` is already an empty string in this case.